### PR TITLE
Implement signed 64-bit shift right for wasm2asm

### PR DESCRIPTION
**Pull request by alexcrichton**
Mostly piggy-back pon the previous 64-bit shift lowering code, just filling in a
few gaps.

### CODE REVIEW

File: src/passes/I64ToI32Lowering.cpp [hunk 1]
```diff
@@ -1042,7 +1101,7 @@ struct I64ToI32Lowering : public WalkerPass<PostWalker<I64ToI32Lowering>> {
 
   Block* lowerShU(BinaryOp op, Block* result, TempVar&& leftLow,
```
Comment (kripken) [inline src/passes/I64ToI32Lowering.cpp]: How about changing `lowerShU` to `lowerShift`, as now it handles signed as well?

File: test/wasm2asm/i64-shifts.wast [hunk 2]
```diff
@@ -0,0 +1,81 @@
+;; Testing i64 shifts
```
Comment (kripken) [inline test/wasm2asm/i64-shifts.wast]: Please run `./auto_update_tests.py` and add the output file it creates. That will be used by the test suite going forward.

**Comment by alexcrichton** [comment]
Thanks! I've renamed the lowering function (good catch!) although after running `./auto_update_tests.py` I'm not seeing any new files to commit. Where should I be expecting such a file to materialize?

**Comment by kripken** [comment]
Hmm, for some reason the script special-cases wasm2asm and requires that the output file exist (same base name, suffix `.2asm.js`). We should probably remove that (lines 302-303). For this PR, you can either remove those 2 lines or create an empty file there to work around that.

**Comment by alexcrichton** [comment]
Hm I still can't seem to figure this out :(

If I touch the `test/wasm2asm/i64-shifts.2asm.js` file or if I comment out [these lines](https://github.com/WebAssembly/binaryen/blob/e4927328ed2f748bd02d47a281d35fe22dfe5ef8/auto_update_tests.py#L302-L303) the `*.js` file doesn't get filled in. It also looks like there's no other `*.2asm.js` files in `test/wasm2asm`?

**Comment by kripken** [comment]
Sorry, I'm not familiar with that code and it is indeed confusing.

For some reason it seems to want to put the test outputs in `/test/` (we should probably fix that), so you need to create that file one directory above. And indeed that is where the `.2asm.js` tests are. Very odd...

**Comment by alexcrichton** [comment]
That seems to have done the trick!

**Comment by kripken** [comment]
So I must apologize again, but the code in this area really needs improvement. Now that I see how much I'll make time to read it and clean it up. It wants the test file under `test/wasm2asm` but the output file goes in the outer dir. And if you put the test file in the outer dir, it does a lot more with it. tl;dr to fix your test failures here, move just the test file, not the output, into `test/wasm2asm`.

**Comment by alexcrichton** [comment]
Let's see how Travis likes that...

**Comment by kripken** [comment]
Worked, thanks!

And sorry again, those testing annoyances are really not how one's first PR should go in any project. I'll look into cleaning that up.

**Comment by alexcrichton** [comment]
Ah no worries! Hopefully this'll make the next patch easier :)

**Pull request merged**